src/Extension/ExtensionConstants.php

```diff
@@ -0,0 +1,15 @@
+<?php namespace Winter\Storm\Extension;
+
+/**
+ * Class ExtensionConstants
+ * Holds the constant values for use in the traits, to limit the use of magic strings
+ * @package Winter\Storm\Extension
+ */
+class ExtensionConstants
+{
+    const EXTENSIONS = 'extensions';
+    const METHODS = 'methods';
+    const FIELDS = 'fields';
+    const DYNAMIC_METHODS = 'dynamicMethods';
+    const DYNAMIC_PROPERTIES = 'dynamicProperties';
+}
```

Comment:
I don't think this is worth the added verbosity to the code. What are your thoughts?

Reply:
Magic strings should be avoided when possible IMO. A magic string is subject to human error, and requires a lot of copy pasting, will not be error checked.

For some of the other reasons check

https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings

Reply:
Now that I think about it, it might even be better to replace the array access used in the extender with management classes so it becomes more self documenting instead of the nested array properties? Your thoughts?

Reply:
I would have to disagree in this specific case. The Extendable functionality is already quite complex, so adding more complexity to it doesn't make it better. Also, the magic strings are for an internal API that we do not want to be encouraging developers to be touching at all, so personally I'm fine with leaving them as magic strings because it's not like it's a public API that should be interacted with by developers on a regular basis.

Reply:
It's changed back now.